### PR TITLE
docs(appengine): document container image url deployments to appengine flex custom

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: setup/install/providers/appengine.md [hunk 1]
```diff
@@ -98,13 +98,31 @@ You can omit the `--json-path` flag if Spinnaker does not need service account c
...
+by deploying a docker container from a gcr.io URL.  If your application is deployed from source code and
+your source code repositories are private then Spinnaker will need a mechanism to authenticate with your
+repositories.  Many of the configuration flags for App Engine manage this authentication.
 
 You can view the available configuration flags for App Engine within the
 [Halyard reference](/reference/halyard/commands#hal-config-provider-appengine-account-add).
 
+## Deploying from Google Container Registry URL
+
+Spinnaker supports deploying docker containers on the Appengine Flex runtime from images built and stored
```
Comment (danielpeach) [inline setup/install/providers/appengine.md]: s/docker/Docker, s/Appengine/App Engine

File: setup/install/providers/appengine.md [hunk 2]
```diff
@@ -98,13 +98,31 @@ You can omit the `--json-path` flag if Spinnaker does not need service account c
...
+in Google Container Registry from just a gcr.io URL.  This feature is currently flagged because it is still quite new.
+
+In order to enable this feature, set the flag with halyard like so:
```
Comment (danielpeach) [inline setup/install/providers/appengine.md]: s/halyard/Halyard

**Review by danielpeach — APPROVED** [review]

**Review by avh121 — COMMENTED** [review]
So, if I'm understanding correctly, users now have two choices for deploying to App Engine: 1) deploying from a git repo, or 2) deploying from GCR. If that's correct, let's refactor the 'Advanced Account Settings' section to have two sub-headings: 'Deploying from GitHub' (which can talk about the authentication mechanisms, etc), and 'Deploying from Google Container Registry URL'.

It also seems like users have to pick one of these two options in order to deploy to App Engine. Is that correct? If so, that seems like a necessary step rather than an advanced setting to me, in which case let's rename 'Advanced Account Settings' to something like 'Deploy to App Engine' above the GitHub/GCR sub-options.

**Comment by danielpeach** [comment]
Should be three paths actually:

- Deploy from a git repository (not necessary Github)
- Deploy from a storage bucket (I think it has to be GCS, but not strictly in the future, though who would deploy to GAE from S3?)
- Deploy from a container image.

**Comment by ghost** [comment]
okey dokey.  i've renamed "Advanced Account Settings" to "Deploying to App Engine" and refactored into two separate sections for deploying from source and deploying from gcr url.

I'll rename "Deploy from Source" to "Deploy from Git" and add a section for deploying from a bucket.

File: setup/install/providers/appengine.md [hunk 3]
```diff
@@ -96,7 +96,9 @@ hal config provider appengine account add my-appengine-account \
 
 You can omit the `--json-path` flag if Spinnaker does not need service account credentials.
 
-## Advanced Account Settings
+## Deploying to App Engine
+
+### Deploying from Source
 
 Spinnaker deploys to App Engine by cloning your application source code from a git repository. Unless your code
```
Comment (lwander) [inline setup/install/providers/appengine.md]: I think the wording here should match that of the below section, e.g. "Spinnaker supports deploying your source code by cloning a git repo and submitting it to App Engine..."

**Review by lwander — APPROVED** [review]

**Comment by ghost** [comment]
I've added a section now for deploying from GCS.  It briefly describes what Spinnaker "does" during that kind of deployment.  I've also reworded the first section on deploying from Git to match wording with the section on deploying from container image URL.

**Review by avh121 — APPROVED** [review]

**Pull request merged**